op-dispute-mon/mon/bonds/monitor.go

```diff
@@ -30,4 +40,80 @@ func (b *Bonds) CheckBonds(games []*types.EnrichedGameData) {
 	for addr, collateral := range data {
 		b.metrics.RecordBondCollateral(addr, collateral.Required, collateral.Actual)
 	}
+
+	for _, game := range games {
+		b.checkCredits(game)
+	}
+}
+
+func (b *Bonds) checkCredits(game *types.EnrichedGameData) {
+	addresses := make([]common.Address, len(game.Claims))
+	credits := make(map[common.Address]*big.Int)
+
+	// Check if the max duration has been reached for this game
+	duration := uint64(b.clock.Now().Unix()) - game.Timestamp
+	maxDurationReached := duration >= game.Duration
+
+	// Iterate over all *resolved* claims
+	claims := b.filterResolvedClaims(game)
+	for _, claim := range claims {
+		// The recipient of a resolved claim is the claimant unless it's been countered.
+		recipient := claim.Claimant
+		if claim.CounteredBy != (common.Address{}) {
+			recipient = claim.CounteredBy
+		}
+		// Append the recipient to our address list if it's not already tracked
+		credit, ok := credits[recipient]
+		if !ok {
+			addresses = append(addresses, recipient)
+		}
+		// TODO: fetch the required bond from the game contract
```

Comment:
Fetching really should be done as part of the extract stage rather than here.  Possibly need the `EnrichedGameData.Claims` array to be an `EnrichedClaim` type (that extends `faultTypes.Claim`) and we can then add an enricher to pull the required bonds for any claims that have been resolved.

As a side benefit would could also add a `Resolved bool` flag to the `Claim` so we don't have to compare the bond to a sentinel value in multiple places through dispute-mon.

Reply:
Yep done, I just pulled those changes out and merged in through a separate pr. Good callout on the resolved flag.